api/v1alpha1/trafficsplit_types.go

```diff
@@ -45,6 +34,126 @@ type TrafficSplit struct {
 	Status TrafficSplitStatus `json:"status,omitempty"`
 }
 
+// TrafficSplitSpec defines the desired state of TrafficSplit
+type TrafficSplitSpec struct {
+	// Destinations defines a schema for splitting traffic to
+	// backend referenced objects.
+	//
+	// Support: core
+	Destinations []TrafficSplitDestination `json:"destinations"`
+}
+
+// TrafficSplitDestination defines the destination and rules for
+// splitting traffic.
+type TrafficSplitDestination struct {
+	// DestinationRef is a reference to an object that should be used
+	// to forward traffic.
+	//
+	// Support: core
+	//
+	DestinationRef core.ObjectReference `json:"destinationRef"`
+	// Percent specifies the split percentage, a number between 0 and 100.
+	// The sum of percentages across destinations must equal 100.
+	// If only one trafficsplit is specified, the default is 100.
+	//
+	// Support: core
+	//
+	// +optional
+	Percent *int `json:"percent,omitempty"`
+	// Filter defines what filters are applied to the request
+	// before forwarding traffic to the destination.
+	//
+	// Support: extended
+	//
+	// +optional
+	Filter *TrafficSplitFilter `json:"filter,omitempty"`
```

Comment:
__TODO:__ Should `Filter` be invalid for passthrough tls connections?

Reply:
I think yes. This should be valid only for HTTP routes. `Filter` should not be interpreted for TCP routes all together, no matter it is plain-text or TLS pass through.